### PR TITLE
serialize isEnabled for lights and cameras

### PR DESCRIPTION
**Pull request by andreasplesch**
for discussion see https://forum.babylonjs.com/t/light-serialization/25036

https://www.babylonjs-playground.com/#PM0XL0 has a few simple tests for light.
I could generate similar tests for camera. `camera.setEnabled(false)` does not seem to have an effect.

I have build artefacts if useful.

Even with only a few lines, here are some questions:

- Remove/add comments ?
- Use of `in` operator ? It does not seem common in the code base but is supported by IE11.
- replace `if` with ternary ? 
```js
node.setEnabled(
    "isEnabled" in parseNode ? parsedNode.isEnabled : true
)
```

### CODE REVIEW

**Comment by sebavan** [comment]
I guess you could do smthg like this
```
if (parsedCamera.isEnabled !== undefined) {
    camera.setEnabled(parsedCamera.isEnabled);
}
```

You do not need comments in the serialized part as it is expressive enough, but might be good to add one in the parse section to let the next dev know the undefined check is for back compat.

**Comment by andreasplesch** [comment]
> ```js
> // Check if isEnabled is defined to be back compatible with prior serialized versions.
> if (parsedCamera.isEnabled !== undefined) {
>     camera.setEnabled(parsedCamera.isEnabled);
> }
> ```
> 
> You do not need comments in the serialized part as it is expressive enough, but might be good to add one in the parse section to let the next dev know the undefined check is for back compat.

Makes sense. I adopted these and added a what's new line.

**Comment by andreasplesch** [comment]
Here is a test scene: https://www.babylonjs-playground.com/#PM0XL0#2

It tests serialization and parsing, also for the back compat. case.

File: src/Cameras/camera.ts [hunk 1]
```diff
@@ -1379,6 +1381,14 @@ export class Camera extends Node {
             scene.beginAnimation(camera, parsedCamera.autoAnimateFrom, parsedCamera.autoAnimateTo, parsedCamera.autoAnimateLoop, parsedCamera.autoAnimateSpeed || 1.0);
         }
 
+        // Check if isEnabled is defined to be back compatible with prior serialized versions.
+        if (parsedCamera.isEnabled !== undefined) {
+            camera.setEnabled(parsedCamera.isEnabled);
+        }
+        else {
+            camera.setEnabled(true);
```
Comment (deltakosh) [inline src/Cameras/camera.ts:1389]: not required as the node will be enabled by default

File: src/Lights/light.ts [hunk 2]
```diff
@@ -710,6 +712,14 @@ export abstract class Light extends Node implements ISortableLight {
             scene.beginAnimation(light, parsedLight.autoAnimateFrom, parsedLight.autoAnimateTo, parsedLight.autoAnimateLoop, parsedLight.autoAnimateSpeed || 1.0);
         }
 
+        // Check if isEnabled is defined to be back compatible with prior serialized versions.
+        if (parsedLight.isEnabled !== undefined) {
+            light.setEnabled(parsedLight.isEnabled);
+        }
+        else {
+            light.setEnabled(true);
```
Comment (deltakosh) [inline src/Lights/light.ts:720]: same

**Review by deltakosh — CHANGES_REQUESTED** [review]
all good. Simpy one line of code that could be removed

File: src/Lights/light.ts (hunk 2 above)
Comment (andreasplesch) [reply src/Lights/light.ts:720]: Right, the back compat. case would be covered by default.

**Pull request merged**